### PR TITLE
pythonPackages.tensorflow 1.1 -> 1.3

### PR DESCRIPTION
**Pull request by jyp**
###### Motivation for this change

Upstream update

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [x] Linux
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

**Comment by akamaus** [comment]
@jyp  what happens to tensorboard after merging this? AFAIK it was separated from tensorflow-1.3. I'm afraid we need a separate expression for it.

**Comment by jyp** [comment]
@akamus: This is correct: this PR removes tensorboard. However, unfortunately, the dependencies of tensorboard do not build at the moment (see the commit). So the PR is not a regression.

**Comment by akamaus** [comment]
@jyp If you're speaking about werkzeug, then there're good news :) Recently I added python3.6 support to tensorflow expression, it got merged  6abb29f460303427d693cb435b2c686477a0865d. cffi works normally on 3.6  so now tensorflow-1.1 is buildable again. Maybe it's better to rebase your work against that commit?

**Comment by jyp** [comment]
@akamaus It does not look like your commit made it to master --- regardless I suppose that if werkzeug is working with python 3.6 then the best is to drop support for 3.5 and go for 3.6 and add tensorboard properly. I'll give it a shot.

**Comment by jyp** [comment]
In fact, tensorboard still won't build, because it depends on exactly htlm5lib 9999999 (that's seven nines), which does not build. If you know @akamaus where to complain to get tensorboard people to put decent dependencies on their lib then let me know.

I'll reiterate my observation that tensorflow works perfectly well without tensorboard. So I maintain that this PR is the way to go until tensorboard uses saner dependencies.

**Comment by akamaus** [comment]
oh, I see now, https://github.com/tensorflow/tensorboard/blob/master/tensorboard/pip_package/setup.py

At first I thought we should file issue there... But look,  nixpkgs (I looked at 17.09) have weird version too
```
  html5lib = buildPythonPackage (rec {
    version = "0.999999999";
    name = "html5lib-${version}";

    src = pkgs.fetchurl {
      url = "http://github.com/html5lib/html5lib-python/archive/${version}.tar.gz";
      sha256 = "09j6194f5mlnd5xwbavwvnndwl1x91jw74shxl6hcxjp4fxg3h05";
    };
```
Please, note, there are different number of nines ;)
I have no experience with python packaging, so I can't comment on this practice.

**Comment by jyp** [comment]
Well the terrible html5lib versioning scheme does not help with typos and misunderstandings of all kind. But the crux of the matter seems to be that tensorboard people are reluctant to upgrade to bleach 2.0.

https://github.com/tensorflow/tensorboard/issues/427

And I'm not to thrilled at the idea of packaging essentially unmaintained packages.

So, to re-interate, because tensorboard is not an essential component of ten, I am dropping it until they eventually move to upgrade. (Also they claim that they don't even support python 3.6; see above issue).

**Comment by abbradar** [comment]
FWIW I have a working build with TensorBoard (which I made separately) but it doesn't support Python 3.6, so I guess your approach is better. 

6abb29f460303427d693cb435b2c686477a0865d has been merged recently which messes with your patch; you probably can just revert it and leave only 2.7 and 3.6.

**Comment by abbradar** [comment]
Giving more thought to this, can you instead make your package support both 3.5 and 3.6 like it currently does on master? We'll then merge my tensorboard on top of your update, so people can have it with tensorboard on 3.5 and without on 3.6.

**Comment by jyp** [comment]
@abbradar I'm not too fond of having to track too many wheel versions, in honesty I won't have the time to test them. (Is there even a good way to automatically update the hashes when a new version comes out?)

If you have a working tensorboard please submit it as a separate PR on top of this one.

**Comment by abbradar** [comment]
@jyp There are some projects to automate this but I haven't tried them yet.

For some reason I can't submit a pull request to you -- GitHub doesn't allow me to select your branch as a base fork. Please check out https://github.com/abbradar/nixpkgs/tree/tf1.3

**Comment by jyp** [comment]
@abbradar Thanks a lot; hopefully this PR will be merged soon (I suppose code owners are busy with the upcoming release right now) and then we can proceed with yours.

@FRidh any chance to merge this PR as it seems that we're reaching a consensus on the way forward?

**Comment by abbradar** [comment]
@jyp Notice that currently this PR is broken because of commit adding Python 3.5 support on master! You have `py3` and `py36` attributes with `py36` being old TensorFlow.

**Comment by jyp** [comment]
@abbradar I am pretty sure that I rebased it. ("This branch has no conflicts with the base branch")

File: pkgs/development/python-modules/tensorflow/default.nix [hunk 1]
```diff
@@ -47,21 +46,21 @@ buildPythonPackage rec {
...
-            sha256 = "0ld3hqx3idxk0zcrvn3p9yqnmx09zsj3mw66jlfw6fkv5hznx8j2";
+            sha256 = "09pcyx0yfil4dm6cij8n3907pfgva07a38avrbai4qk5h6hxm8w9";
           };
-          py35 = {
-            url = tfurl "linux" "cpu" "cp35-cp35m-linux_x86_64";
-            sha256 = "0ahz9222rzqrk43lb9w4m351klkm6mlnnvw8xfqip28vbmymw90b";
+          py3 = {
+            url = tfurl "linux" "cpu" "cp36-cp36m-linux_x86_64";
+            sha256 = "1qm8lm2f6bf9d462ybgwrz0dn9i6cnisgwdvyq9ssmy2f1gp8hxk";
           };
```
Comment (abbradar) [inline pkgs/development/python-modules/tensorflow/default.nix]: @jyp The problem is for example here, notice `py3` and `py36` are both in your result. Git still managed to merge it without conflicts ;)

**Comment by jyp** [comment]
Alright; the PR was not in fact broken but had some dead code. Fixed anyway :)

**Comment by abbradar** [comment]
Yeah, I used too strong a word. I have tested this locally (not on GPUs) and it works for me. Will merge in several days unless anybody objects.

Thank you for your work!

**Comment by abbradar** [comment]
Merged along with my support for 3.5 in 01d53df77f26d0e87fea568c612f3aec8280a266. Tested on a simple VAE, worked for me.

**Pull request closed without merging**